### PR TITLE
Missing documentation

### PR DESCRIPTION
**Pull request by tarrencev**
Adds documentation for `BoxTask` and a couple other minor fixes

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tarrencev has signed the CLA. Thanks for the pull request!

File: doc/tasks.md [hunk 1]
```diff
@@ -1,4 +1,103 @@
...
+Update a Task's Information
+--------------------------
+
+The message of a task can be changed with the
+[`updateInfo(String)`][update-info] method.
+
+```java
+BoxTask task = new BoxTask(api, "id");
+BoxTask.Info info = task.new Info();
+info.setMessage("An edited message.");
```
Comment (tarrencev) [inline doc/tasks.md]: `BoxTask.Info` currently does not expose any setters, so this example wouldn't actually work. I'll expose the necessary setters in a different PR

File: doc/tasks.md [hunk 2]
```diff
@@ -1,4 +1,103 @@
 Tasks
 =====
 
-Docs coming soon.
+Task objects represent a user-created tak on a file.
```
Comment (gcurtis) [inline doc/tasks.md]: Typo

File: doc/tasks.md [hunk 3]
```diff
@@ -1,4 +1,103 @@
...
+
+* [Get a Task's Information](#get-a-tasks-information)
+* [Get the Tasks on a File](#get-the-tasks-on-a-file)
+* [Add a Task to a File](#add-a-task-to-a-file)
+* [Update a Task's Information](#update-a-tasks-information)
+* [Get a Task's Assignments](#get-a-tasks-assignments)
+* [Add a Task Assignment](#add-a-task-assignment)
+
+Get a Task's Information
+---------------------------
```
Comment (gcurtis) [inline doc/tasks.md]: Match the length of the underline with the length of the header.

**Comment by gcurtis** [comment]
Thanks! Looks really good, just a couple minor comments.

**Pull request merged**